### ISSUE TITLE
Make an in-browser version

### ISSUE THREAD

**Issue by louischatriot** [louischatriot/nedb#23]
After seeing how much faster NeDB is than TaffyDB, I'm considering porting NeDB to the browser, with persistence support using LocalStorage. Please comment if you think this would be useful.

**Comment by louischatriot** [comment]
As of v1.4, NeDB automatically selects the best method to persist data on the browser (IndexDB then WebSQL then localStorage), so the limit is removed. WARNING: it's not back compatible with v1.3, but that should not be a major issue as user apps can resync entirely.

**Comment by DasAnudas** [comment]
NeDB for browsers and mobile platforms like Andoid, ios through Phonegap to persist data locally will be utmost useful utility database. I am sure it will be the most popular database after Mongodb.  Though, I tried including version 1.8, as mentioned in package.json, of nedb.js and/or nedb.min.js in html file using script tag from folder nedb/ browsers-version/out folder, but it gives error.  Further guidance for how to include nedb.js in html file for saving data on local disk will be very much appreciated...

<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Document</title>

  
  <link rel="stylesheet" href="css/material.css">
  <script src="js/material.js"></script>

  
  <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
  <link rel="stylesheet" href="https://code.getmdl.io/1.3.0/material.min.css">

  <style>
    body {
      margin-left: 15px;
    }
    input {
      width: 400px;
      margin-bottom: 5px;
    }

    label {
      width: 90px;
      /* text-align: right;  */
       float: left; 
    }

    form {
      margin-left: 5px;
    }

  </style>
</head>
<body>
  <div id="container">
    
    <div id="inputs">
      <h4>Personal Details:</h4>

      <form name="testForm"
      action="#"
      method="POST">
      
        <label for="fname">First Name:</label>
        <input type="text" placeholder="First name" id="fname"><br>
        <label for="mname">Middle Name:</label>
        <input type="text" placeholder="Middle name" id="mname"><br>
        <label for="lname">Last Name:</label>
        <input type="text" placeholder="Last name" id="lname"><br>
        <label for="building">Building:</label>
        <input type="text" placeholder="Building" id="building"><br>
        <label for="street">Street:</label>
        <input type="text" placeholder="Street" id="street"><br>
        <label for="locality">Locality:</label>
        <input type="text" placeholder="Locality" id="locality"><br>
        <label for="landmark">Landmark:</label>
        <input type="text" placeholder="Landmark" id="landmark"><br>
        <label for="city">City:</label>
        <input type="text" placeholder="City" id="city"><br>
        <label for="zip">Zip/Pin Code:</label>
        <input type="text" placeholder="ZIP/PIN Code" id="zip"><br>
        <label for="state">State:</label>
        <input type="text" placeholder="State" id="state"><br>
        <label for="country">Country:</label>
        <input type="text" placeholder="Country" id="country"><br>  
      </form>
    </div>

    
    <div id="buttons">
      
      <button class="mdl-button
        mdl-js-button
        mdl-button--raised 
        mdl-js-ripple-effect 
        mdl-button--accent"
        type="submit"
        onclick="saveContact();">
        Save Contact
      </button>

      <hr>

      <h4>Result:</h4>
      <p id="result"></p>

      
      
    </div>
  </div>

  
  
  <script src="lib/mdl/1.3.0/material.js"></script> 

  
  

  
   

  
  <script src="lib/nedb/browser-version/out/nedb.min.js"></script>
  

  
  
  

  
  <script src="js/index.js"></script>
</body>
</html>

**Comment by marcusjwhelan** [comment]
@DasAnudas there is actually a db inspired by nedb but built on in many ways and updated to Es6 and has the capability to do all the things you would like. you could write an html parser to overhead this db as well. It's maintained by myself and another and has others who are watching and waiting for new ideas to build on. https://github.com/tedb-org/teDB

**Comment by DasAnudas** [comment]
Hi there, to all of you developers. Thanks of letting me know about
your new development.

I downloaded teDB from link given and had a look.  But I did not find
a way to include teDB in HTML file using script tag. So I don't know,
if it can solve my purpose of saving HTML forms data locally.

I was using function Datastore() of NeDB to create a datastore was
giving error. But when I saw index.html file in browser-version/test
which used function Nedb() to create datasotre. I adapted this change
in my script and no more error now. But this function is not creating
db file on local disk is my problem at the moment. I used following
code:
"contacts = new Nedb({ filename: "contacts.db", autoload: true });"

Further, I appreciate your valuable guidance to both issues.

best regards,
DasAnudas.
